### PR TITLE
docs: Elaborate on SubDomains for BCs w MPI

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: examples/userapi/04_boundary_conditions.ipynb [hunk 1]
```diff
@@ -11,7 +11,10 @@
    "cell_type": "markdown",
```
Comment (EdCaunt) [inline examples/userapi/04_boundary_conditions.ipynb:11]: <p>Typo: "user code"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/devitocodes/devito/pull/2491/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: devito/types/dense.py [hunk 2]
```diff
@@ -1399,7 +1402,7 @@ def __indices_setup__(cls, *args, **kwargs):
 
             if time_dim is None:
                 time_dim = grid.time_dim if isinstance(save, int) else grid.stepping_dim
-            elif not (isinstance(time_dim, Dimension) and time_dim.is_Time):
+            elif isinstance(time_dim, TimeDimension):
```
Comment (mloubout) [inline devito/types/dense.py:1405]: `not`

Comment (georgebisbas) [reply devito/types/dense.py:1405]: not Dimension and time_dim.is_Time

equivalent to 

TImeDimension

right?

Comment (mloubout) [reply devito/types/dense.py:1405]: not Dimension and is_Time is impossible, check the parentheses it's not ( Dimension and tine)

Comment (georgebisbas) [reply devito/types/dense.py:1405]: Sorry, yes. Interesting that no test failed! Reverted, as it is not semantically equivalent

File: examples/userapi/04_boundary_conditions.ipynb (hunk 1 above)
Comment (georgebisbas) [reply examples/userapi/04_boundary_conditions.ipynb:11]: fixed, thanks

**Pull request merged**